src/providers/wms/qgswmsprovider.cpp

```diff
@@ -1130,11 +1130,20 @@ void QgsWmsProvider::createTileRequestsXYZ( const QgsWmtsTileMatrix *tm, const Q
       turl.replace( QLatin1String( "{q}" ), _tile2quadkey( tile.col, tile.row, z ) );
 
     turl.replace( QLatin1String( "{x}" ), QString::number( tile.col ), Qt::CaseInsensitive );
-    turl.replace( QLatin1String( "{y}" ), QString::number( tile.row ), Qt::CaseInsensitive );
+    // inverted Y axis
+    bool invertedY = turl.contains( QLatin1String( "{-y}" ) );
+    if ( invertedY )
+    {
+      turl.replace( QLatin1String( "{-y}" ), QString::number( tm->matrixHeight - tile.row ), Qt::CaseInsensitive );
```

Comment:
Can we change that expression to tm->matrixHeight - tile.row - 1 ? That would avoid the need to have extra argument in tileRect() method, making the code simpler and easier to understand...